### PR TITLE
Fix Bloop project generation in buildall [skip ci]

### LINKED ISSUE #5706: [BUG] buildall --generate-bloop creates projects that Metals/Bloop does not recognize in VS code
**Describe the bug**
The logic in --generate-bloop creating spark-version-classified Bloop projects no longer generate compilable projects. 

The intent of jq post processing is to allow multiple projects with different spark dependency 
like spark-rapids_2.12-spark321 in the same .bloop dir. I no longer think it has any practical value, and it's definitely broken even if we use a single buildver when running `buildall -gb --profiles=321`

**Steps/Code to reproduce bug**
buildall -gb --profiles=321 

follow instructions in CONTRIBUTING.md for VS Code, and observe we can no longer activate the Metals project

**Expected behavior**
Should work

**Environment details (please complete the following information)**
local

**Additional context**
Easy workaround to just run bloopInstall directly 
```bash
mvn install ch.epfl.scala:maven-bloop_2.13:1.4.13:bloopInstall \
  -Dbuildver=321 \
  -DdownloadSources=true \
  -Dskip -Dmaven.javadoc.skip \
  -DskipTests
```

### PR DESCRIPTION
**Pull request by gerashegalov**
This PR fixes #5706

- Generates a Bloop project for each Spark dependency specified via `--profile`
- Explains how to activate a particular project
- No longer tries to have multiple Spark dependencies in the same .bloop
directory
- Fixes incorrect BUILD_ALL_DEBUG tests
- Fixes match for .bloop in gitignore 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -218,9 +218,21 @@ not clobbered by repeated `bloopInstall` Maven plugin invocations, and it uses
 [jq](https://stedolan.github.io/jq/) to post-process JSON-formatted project files such that they
 compile project classes into non-overlapping set of output directories.
 
+To activate the Spark dependency you currentky work update the symlink
```
Comment (amahussein) [inline CONTRIBUTING.md:221]: typo. currentky -> currently

```
The script creates a symlink `.bloop` that points to the active Bloop project relative to the profile parameter.  
To activate the Spark dependency to a specific version, you need to update the symlink...
```

**Review by amahussein — COMMENTED** [review]
Just a minor suggestion.

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Review by amahussein — APPROVED** [review]
LGTM

**Pull request merged**